### PR TITLE
Fix highlight main nav menu items

### LINKED ISSUE #10010: App bar does not highlight current cluster when product is not explorer
**Setup**
- Rancher version:
- Rancher UI Extensions:
- Browser type & version:

**Describe the bug**
- The app bar cluster icon should be highlighted when the user is in the cluster explorer context
- When navigating to a non explorer product part of cluster explorer the cluster icon highlight is removed

**To Reproduce**
- Install a chart to a cluster, suggest something simple like CIS Benchmark
- Navigate to the CIS Benchmark side nav menu group

**Result**
- The highlight of the selected cluster is removed

**Expected Result**
- Highlight remains

**Screenshots**
Correct state
image

Incorrect state
image

**Additional context**
- state works off of nuxt's `nuxt-link-active` class
- this works from the c-cluster-explorer route, which is product specific
- so going to other products in the explorer view fails, so something like c-cluster-cis

### PR DESCRIPTION
**Pull request by aalves08**
### Summary
Fixes #10010 

### Occurred changes and/or fixed issues

- Add method `checkActiveRoute` to check and apply a class `active-menu-link` to main nav menu items
- Update Fleet product route to enforce the use of `blank` cluster
- Update e2e tests in key places to check if main nav menu items are highlighted correctly when navigating to sub-routes
- Update e2e tests for Fleet to navigate to it's routes using `blank` cluster

### Areas or cases that should be tested

- Generally test navigation to all the different products Rancher Dashboard has to offer

### Screenshot/Video

Screenshot 2024-01-03 at 12 30 45

### CODE REVIEW

**Review by richard-cox — CHANGES_REQUESTED** [review]

File: cypress/e2e/tests/pages/charts/istio.spec.ts [hunk 1]
```diff
@@ -22,6 +23,9 @@ describe('Charts', { tags: ['@charts', '@adminUser'] }, () => {
         chartsPage.goTo();
         chartsPage.goToInstall().nextPage();
 
+        // check if burguer menu nav is highlighted correctly for local cluster (apps is a different prod than explorer)
+        BurgerMenuPo.burgerMenuGetNavClusterbyLabel('local').parent().should('have.class', 'active-menu-link');
```
[the remark this answers is not in the document]
Comment (aalves08) [reply cypress/e2e/tests/pages/charts/istio.spec.ts:27]: updated

File: shell/components/nav/TopLevelMenu.vue [hunk 2]
```diff
@@ -273,6 +274,18 @@ export default {
       return `${ lineHeightInEm }em`;
     },
 
+    checkActiveRoute(obj, isClusterRoute) {
+      // cluster level routes (EXPLORER, APPS, etc)
+      if (isClusterRoute && this.$route?.name?.startsWith('c-cluster') && this.currentCluster?.id === obj?.id) {
+        return this.$route?.name?.startsWith('c-cluster') && this.$route?.params?.cluster === obj?.id && getProductFromRoute(this.$route) === this.currentProduct?.name;
```
[the remark this answers is not in the document]
Comment (aalves08) [reply shell/components/nav/TopLevelMenu.vue:280]: 🤦 fixed

File: shell/components/nav/TopLevelMenu.vue [hunk 3]
```diff
@@ -273,6 +274,18 @@ export default {
       return `${ lineHeightInEm }em`;
     },
 
+    checkActiveRoute(obj, isClusterRoute) {
+      // cluster level routes (EXPLORER, APPS, etc)
+      if (isClusterRoute && this.$route?.name?.startsWith('c-cluster') && this.currentCluster?.id === obj?.id) {
```
[the remark this answers is not in the document]
Comment (aalves08) [reply shell/components/nav/TopLevelMenu.vue]: updated with comments and simplified logic

**Review by richard-cox — APPROVED** [review]

File: cypress/e2e/tests/pages/charts/istio.spec.ts [hunk 4]
```diff
@@ -22,6 +23,9 @@ describe('Charts', { tags: ['@charts', '@adminUser'] }, () => {
         chartsPage.goTo();
         chartsPage.goToInstall().nextPage();
 
+        // check if burguer menu nav is highlighted correctly for local cluster (apps is a different prod than explorer)
+        BurgerMenuPo.checkIfClusterMenuLinkIsHighlighted('local');
```
Comment (izaac) [inline cypress/e2e/tests/pages/charts/istio.spec.ts:27]: There is an existging `clusterName` variable on the top describe

**Pull request merged**